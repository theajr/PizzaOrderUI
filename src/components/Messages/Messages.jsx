import React from 'react';
import Snackbar from '@material-ui/core/Snackbar';
import { connect } from '../../util/appContext';
import { REMOVE_MESSAGE } from '../../util/constants';

const Messages = props => {
  return props.errors.map(err => (
    <Snackbar
      key={err.id}
      open={true}
      onClose={(e, r) => {
        props.removeMessage(err.id);
      }}
      // anchorOrigin={{ vertical: 'top', horizontal: 'right' }}
      autoHideDuration={2000}
      ContentProps={{}}
      message={<span>{err.message}</span>}
    />
  ));
};

const mapStateToProps = state => {
  return {
    errors: state.errors || [],
  };
};

const mapDispatchToProps = dispatch => {
  return {
    removeMessage: msg => dispatch({ type: REMOVE_MESSAGE, payload: msg }),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Messages);
