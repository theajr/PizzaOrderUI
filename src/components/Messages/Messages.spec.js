import React from 'react';
import { shallow } from 'enzyme';
import Messages from './Messages';
import { renderComp } from '../../../tests/utilRender';

describe('<Messages />', () => {
  test('renders', () => {
    const wrapper = renderComp(<Messages />);
    expect(wrapper).toMatchSnapshot();
  });
});
