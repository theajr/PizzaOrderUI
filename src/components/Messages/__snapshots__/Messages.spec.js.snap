// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Messages /> renders 1`] = `ShallowWrapper {}`;
